addons/website/static/src/interactions/menu/mega_menu_dropdown.js

```diff
@@ -0,0 +1,103 @@
+import { Interaction } from "@website/core/interaction";
+import { registry } from "@web/core/registry";
+
+class MegaMenuDropdown extends Interaction {
+
+    static selector = "header#top";
+    static dynamicContent = {
+        ".o_mega_menu_toggle": {
+            "t-on-mouseenter": "onHoverMegaMenu",
+            "t-on-mousedown": "onTriggerMegaMenu",
+            "t-on-keyup": "onTriggerMegaMenu",
+        },
+        "_root": {
+            "t-on-mousedown": "onTriggerExtraMenu",
+            "t-on-keyup": "onTriggerExtraMenu",
+        },
+    }
+
+    setup() {
+        this.mobileMegaMenuToggleEls = [];
+        this.desktopMegaMenuToggleEls = [];
+        const megaMenuToggleEls = this.el.querySelectorAll(".o_mega_menu_toggle");
+        for (const megaMenuToggleEl of megaMenuToggleEls) {
+            if (megaMenuToggleEl.closest(".o_header_mobile")) {
+                this.mobileMegaMenuToggleEls.push(megaMenuToggleEl);
+            } else {
+                this.desktopMegaMenuToggleEls.push(megaMenuToggleEl);
+            }
+        }
+    }
+
+    /**
+     * If the mega menu dropdown on which we are clicking/hovering does not have
+     * a mega menu (i.e. it is in the other navbar), brings the corresponding
+     * mega menu into it.
+     *
+     * @param {Element} megaMenuToggleEl
+     */
+    moveMegaMenu(megaMenuToggleEl) {
+        const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+        if (hasMegaMenu) {
+            return;
+        }
+        //this.options.wysiwyg?.odooEditor.observerUnactive("moveMegaMenu");
```

Comment:
Maybe add a `// TODO Editor behavior` above such lines, so that we can think about checking that behavior later, when integrating with the new HTML Builder.